Dynavity/Dynavity/model/canvas/Canvas.swift

```diff
@@ -36,25 +40,34 @@ class Canvas: ObservableObject {
             return
         }
 
-        if element is UmlElementProtocol {
-            removeAttachedConnectors(element as? UmlElementProtocol)
+        if let umlElement = element as? UmlElementProtocol {
+            removeAttachedConnectors(umlElement)
         }
         canvasElements.remove(at: index)
         canvasElementCancellables.remove(at: index)
     }
 
-    private func removeAttachedConnectors(_ element: UmlElementProtocol?) {
-        guard let umlElement = element else {
-            return
-        }
+    func replace(canvasElements: [CanvasElementProtocol]) {
+        self.canvasElements
+            .filter { !($0 is UmlElementProtocol) }
+            .forEach(removeElement)
+        canvasElements.forEach(addElement)
+    }
 
-        for connector in umlConnectors {
-            if connector.connects.fromElement !== umlElement
-                    && connector.connects.toElement !== umlElement {
-                continue
-            }
-            removeUmlConnector(connector)
-        }
+    func replace(annotation: AnnotationCanvas) {
+        self.annotationCanvas = annotation
```

Comment:
Perhaps the reason why the annotations changes are not being reflected is because the entire `AnnotationCanvas` is replaced, thus the observers don't work? Could we instead replace the `drawing` within the `annotationCanvas` instead of replacing the entire `annotationCanvas` with a new one?